src/main/java/io/vertx/ext/hawkular/impl/inventory/InventoryReporter.java

```diff
@@ -66,53 +73,48 @@ public InventoryReporter(Vertx vertx, VertxHawkularOptions options, Context cont
       subResourceReporters = new ArrayList<>();
       subResourceReporters.add(new EventbusResourceReporter(options, httpClient));
     });
+    sendTime = System.nanoTime();
+    batchDelay = NANOSECONDS.convert(options.getBatchDelay(), SECONDS);
   }
   public void report() {
     context.runOnContext(aVoid -> {
       Future<Void> feedCreated = Future.future();
       Future<Void> rootResourceCreated = Future.future();
-      Future<Void> subResourcesCreated = Future.future();
       feedReporter.createFeed(feedCreated);
       feedCreated.compose(aVoid1 -> {
         rootResourceReporter.report(rootResourceCreated);
       }, rootResourceCreated);
-      rootResourceCreated.compose(aVoid1 -> {
-        List<Future> futureList = new ArrayList<>(subResourceReporters.size());
-        for (int i = 0; i < subResourceReporters.size(); i++) {
-          EntityReporter r = subResourceReporters.get(i);
-          Future<Void> future = Future.future();
-          r.report(future);
-          futureList.add(future);
-        }
-        CompositeFuture.all(futureList).setHandler(ar -> {
-          if (ar.succeeded()) {
-            subResourcesCreated.complete();
-          } else {
-            subResourcesCreated.fail(ar.cause());
-          }
-        });
-      }, subResourcesCreated);
-      subResourcesCreated.setHandler(ar -> {
-        if (ar.succeeded()) {
-          System.out.println("DONE");
-        } else {
-          System.err.println(ar.cause().getLocalizedMessage());
-        }
+      rootResourceCreated.setHandler(ar -> {
+        timerId = vertx.setPeriodic(MILLISECONDS.convert(batchDelay, NANOSECONDS), this::reportSubResources);
```

Comment:
I think of a solution.
Since I don't know when the callback is called, I just report the resource periodically.